unstructured/partition/text.py

```diff
@@ -0,0 +1,64 @@
+import email
+import re
+from typing import Dict, IO, List, Optional
+
+from unstructured.cleaners.core import replace_mime_encodings
+from unstructured.documents.elements import Element, Text
+from unstructured.partition.text_type import (is_possible_narrative_text
+                                              ,is_possible_title
+                                              ,is_bulleted_text
+                                             )
+
+
+def partition_text(
+    filename: Optional[str] = None, file: Optional[IO] = None, text: Optional[str] = None
+) -> List[Element]:
+    """Partitions an .eml documents into its constituent elements.
+    Parameters
+    ----------
+     filename
+        A string defining the target filename path.
+    file
+        A file-like object using "r" mode --> open(filename, "r").
+    text
+        The string representation of the .eml document.
+    """
+    if not any([filename, file, text]):
+        raise ValueError("One of filename, file, or text must be specified.")
+
+    if filename is not None and not file and not text:
+        with open(filename, "r") as f:
+            msg = email.message_from_file(f)
```

Comment:
These should all deal with plain text `.txt` documents rather than emails. And we'll use it (1) when we want to partition a plain text document and (2) if your partitioning an email and choose to process the `text/plain` content.